### ISSUE TITLE
Extra copy when coercing

### ISSUE THREAD

**Issue by mglisse** [cython/cython#3574]
f.h
```c++
#include <vector>
void f(std::vector<double> const&);
```
g.pyx
```cython
from libcpp.vector cimport vector

cdef extern from "f.h":
    cdef void f(vector[double]) nogil

def g(x):
    f(x)

def h(x):
    cdef vector[double] y = x
    with nogil:
        f(y)
```
For g, cython generates roughly:
```c++
  std::vector<double>  __pyx_t_1;
  __pyx_t_1 = __pyx_convert_vector_from_py_double(__pyx_v_x); if (unlikely(PyErr_Occurred())) __PYX_ERR(0, 7, __pyx_L1_error)
  f(__pyx_t_1);
```
which seems ok. For h, it gives:
```c++
  std::vector<double>  __pyx_v_y;
  std::vector<double>  __pyx_t_1;
  __pyx_t_1 = __pyx_convert_vector_from_py_double(__pyx_v_x); if (unlikely(PyErr_Occurred())) __PYX_ERR(0, 10, __pyx_L1_error)
  __pyx_v_y = __pyx_t_1;
[...]
        f(__pyx_v_y);
```
That's too much copying. I don't think I can avoid naming the variable `y` if I want to use `nogil`. I assume cython doesn't want to write the result of the conversion directly to y in case an error occurs (?). But could it at least generate `__pyx_v_y = MOVE(__pyx_t_1);` where `MOVE` is `std::move` in C++11 or later?
The obvious workaround is to avoid using C++ types in interfaces, stick to `double*` and size, but we don't always choose what we are interfacing...
Cython version 0.29.14

**Comment by scoder** [comment]
Is this solved by #3362? Please try 3.0.

**Comment by mglisse** [comment]
Oups, I just tried master (I should have done that first...) and indeed it is fixed, sorry for the noise.
